devlink_linux.go

```diff
@@ -0,0 +1,166 @@
+package netlink
+
+import (
+	"syscall"
+
+	"github.com/vishvananda/netlink/nl"
+	"golang.org/x/sys/unix"
+)
+
+// DevlinkDevEswitchAttr represents device's eswitch attributes
+type DevlinkDevEswitchAttr struct {
+	Mode       string
+	InlineMode string
+	EncapMode  string
+}
+
+// DevlinkDevAttrs represents device attributes
+type DevlinkDevAttrs struct {
+	Eswitch DevlinkDevEswitchAttr
+}
+
+// DevlinkDevice represents device and its attributes
+type DevlinkDevice struct {
+	BusName    string
+	DeviceName string
+	Attrs      DevlinkDevAttrs
+}
+
+func parseDevLinkDeviceList(msgs [][]byte) ([]*DevlinkDevice, error) {
+	devices := make([]*DevlinkDevice, 0, len(msgs))
+	for _, m := range msgs {
+		attrs, err := nl.ParseRouteAttr(m[nl.SizeofGenlmsg:])
+		if err != nil {
+			return nil, err
+		}
+		dev := &DevlinkDevice{}
+		if err = dev.parseAttributes(attrs); err != nil {
+			return nil, err
+		}
+		devices = append(devices, dev)
+	}
+	return devices, nil
+}
+
+func parseEswitchMode(mode uint16) string {
```

Comment:
Would be great, if possible, to change this into a map that capture all the possible modes:
```
var L4ProtoMap = map[uint16]string{
	nl.DEVLINK_ESWITCH_MODE_LEGACY:  "legacy",
	nl.DEVLINK_ESWITCH_MODE_SWITCHDEV: "switchdev",
        [...]
```

Reply:
@fcrisciani , yes, this is possible, if we dont' find it in map, will continue to return unknown in else condition, so that when tomorrow if kernel adds new option, we can still return unknown.

Reply:
sure